agent/main.go

```diff
@@ -78,6 +78,9 @@ var (
 	metricsbindaddress string
 	downloadpath       string
 	skipInstallation   bool
+	printVersion       bool
+	BuildVersion       string = ""
```

Comment:
Nitpick: Instead of defining a default value in Makefile can it be initialized to `dev` here.

Reply:
The Makefile has to pass some value for this variable as it is part of the build command ldflags. And that value will be overwritten to this variable, even if it has a default value.

The alternative will be to have a conditional block in Makefile where if the BuildVersion isn't set in Makefile we run another build command which doesn't pass this value, but that approach would be less neat.

Reply:
> And that value will be overwritten to this variable, even if it has a default value
yup makes sense. Thanks for correcting me.